lib/api/json_api.ex

```diff
@@ -0,0 +1,191 @@
+defmodule JsonApi.Item do
+  defstruct [:type, :id, :attributes, :relationships]
+
+  @type t :: %JsonApi.Item{
+          type: String.t(),
+          id: String.t(),
+          attributes: %{String.t() => any},
+          relationships: %{String.t() => list(JsonApi.Item.t())}
```

Comment:
I see this is hydrating relationships by pulling their values from the included data. I have some vague memories of hearing other teams regret doing that, and I don't think it'd work for streaming anyway since included resources are only sent once the first time they're seen, so we probably shouldn't.

Reply:
This is all copied directly from dotcom so we can consider future improvements, but it works for streaming.